src/category_theory/limits/types.lean

```diff
@@ -13,8 +13,11 @@ namespace category_theory.limits.types
 
 variables {J : Type u} [small_category J]
 
+def limit_condition (F : J ⥤ Type u) : set (Π j, F.obj j) :=
```

Comment:
If you call it a "condition" isn't it cleaner to make the type `Prop` instead of `set`? That way you can even put `(u) {j j'} (f)` in the assumptions.

Reply:
Indeed. Fixed.

Reply:
Err... I don't see how to move `{j j'} (f)` left of the colon. Nevertheless I made this a `Prop`.

Reply:
Scratch that, it makes proofs gross. I want to leave it as a set...

Reply:
The problem is that I'm using this to argue about subsets being submonoids, and all of that is said in terms of set membership, rather than predicates.

Reply:
Ok, I understand. The name "condition" suggests a predicate. How about renaming this to `compatible_terms`, instead of `limit_condition`? That sounds more set-like...

Reply:
I rearranged this so we just have 
```
def sections (F : J ⥤ Type w) : set (Π j, F.obj j) :=
λ u, ∀ {j j'} (f : j ⟶ j'), F.map f (u j) = u j'
```
in the `functor` namespace, and you just write things like:
```
instance sections_submonoid (F : J ⥤ CommRing.{u}) : is_submonoid (F ⋙ forget).sections :=
```

Reply:
That seems like a very good move :+1: